pkg/controller/stash.go

```diff
@@ -220,27 +228,88 @@ func (c *StashController) getTotalHosts(target interface{}, namespace string) (*
 		t := target.(*api_v1beta1.RestoreTarget)
 		if t == nil {
 			return types.Int32P(1), nil
+		} else if driver == api_v1beta1.VolumeSnapshotter {
```

Comment:
this `driver == api_v1beta1.VolumeSnapshotter` condition has no relation with `t`. So move it to a separate if clause after `targetRef = t.Ref` line.